samples/notebooks/prediction/SDK_Pytorch_Custom_Predict.ipynb

```diff
@@ -0,0 +1,1603 @@
+    "\n",
+    "A [`Predictor`](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/predictor.py) must implement the following interface:\n",
+    "\n",
+    "```\n",
+    "class Predictor:\n",
+    "    \"\"\"Interface for Predictor class that users would be implementing.\"\"\"\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        raise NotImplementedError(\"Predictor.__init__ has not been implemented yet.\")\n",
+    "\n",
+    "    def load(self, gcs_artifacts_uri: str):\n",
+    "        \"\"\"Loads the model artifact.\n",
+    "        Args:\n",
+    "            gcs_artifacts_uri (str):\n",
+    "                Required. The value of the environment variable AIP_STORAGE_URI.\n",
+    "        \"\"\"\n",
+    "        raise NotImplementedError(\"Predictor.load has not been implemented yet.\")\n",
+    "\n",
+    "    def preprocess(self, prediction_input: Any) -> Any:\n",
+    "        \"\"\"Preprocesses the prediction input before doing the prediction.\n",
+    "        Args:\n",
+    "            prediction_input (Any):\n",
+    "                Required. The prediction input needs to be preprocessed.\n",
+    "        Returns:\n",
+    "            The preprocessed prediction input.\n",
+    "        \"\"\"\n",
+    "        return prediction_input\n",
+    "\n",
+    "    def predict(self, instances: Any) -> Any:\n",
+    "        \"\"\"Performs prediction.\n",
+    "        Args:\n",
+    "            instances (Any):\n",
+    "                Required. The instances to perform prediction.\n",
+    "        Returns:\n",
+    "            Prediction results.\n",
+    "        \"\"\"\n",
+    "        raise NotImplementedError(\"Predictor.predict has not been implemented yet.\")\n",
+    "\n",
+    "    def postprocess(self, prediction_results: Any) -> Any:\n",
+    "        \"\"\"Postprocesses the prediction results.\n",
+    "        Args:\n",
+    "            prediction_results (Any):\n",
+    "                Required. The prediction results.\n",
+    "        Returns:\n",
+    "            The postprocessed prediction results.\n",
+    "        \"\"\"\n",
+    "        return prediction_results\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-KODNAqK5k8R"
+   },
+   "source": [
+    "First, implement a custom `Predictor` that loads in both the preprocesor and the model. The preprocessor and the model will then be used at `predict` time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "fkh8DT7CCgQO"
+   },
+   "outputs": [],
+   "source": [
+    "%%writefile $USER_SRC_DIR/predictor.py\n",
+    "\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "import torch\n",
+    "\n",
+    "from google.cloud import storage\n",
+    "from google.cloud.aiplatform.prediction.predictor import Predictor\n",
+    "\n",
+    "\n",
+    "class CprPredictor(Predictor):\n",
```

Comment:
More specific name might be appropriate. CustomPyTorchPredictor?

Reply:
Done.